hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/rejecttokens/RejectTokensTranslatorTest.java

```diff
@@ -0,0 +1,302 @@
+                gasCalculator,
+                configuration);
+
+        // when:
+        boolean matches = subject.matches(attempt);
+
+        // then:
+        assertTrue(matches);
+    }
+
+    @Test
+    void matchesNftHRCWithConfigDisabled() {
+        // given:
+        given(configuration.getConfigData(ContractsConfig.class)).willReturn(contractsConfig);
+        given(contractsConfig.systemContractRejectTokensEnabled()).willReturn(false);
+        given(enhancement.nativeOperations()).willReturn(nativeOperations);
+        attempt = prepareHtsAttemptWithSelectorForRedirectWithConfig(
+                RejectTokensTranslator.HRC_TOKEN_REJECT_NFT,
+                subject,
+                enhancement,
+                addressIdConverter,
+                verificationStrategies,
+                gasCalculator,
+                configuration);
+
+        // when:
+        boolean matches = subject.matches(attempt);
+
+        // then:
+        assertFalse(matches);
+    }
+
+    @Test
+    void gasRequirementCalculatesCorrectly() {
+        long expectedGas = 1000L;
+        given(gasCalculator.gasRequirement(transactionBody, DispatchType.TOKEN_REJECT, payerId))
+                .willReturn(expectedGas);
+
+        long result = RejectTokensTranslator.gasRequirement(transactionBody, gasCalculator, enhancement, payerId);
+
+        assertEquals(expectedGas, result);
+    }
+
+    @Test
+    void callFromHtsTokenReject() {
+        // given:
+        given(addressIdConverter.convertSender(any())).willReturn(SENDER_ID);
+        given(verificationStrategies.activatingOnlyContractKeysFor(any(), anyBoolean(), any()))
+                .willReturn(verificationStrategy);
+        attempt = prepareHtsAttemptWithSelectorAndCustomConfig(
+                RejectTokensTranslator.TOKEN_REJECT,
+                subject,
+                enhancement,
+                addressIdConverter,
+                verificationStrategies,
+                gasCalculator,
+                configuration);
+
+        // when:
+        var call = subject.callFrom(attempt);
+
+        // then:
+        assertEquals(DispatchForResponseCodeHtsCall.class, call.getClass());
+    }
+
+    @Test
+    void callFromHRCCancelFTAirdrop() {
+        // given:
+        given(addressIdConverter.convertSender(any())).willReturn(SENDER_ID);
+        given(verificationStrategies.activatingOnlyContractKeysFor(any(), anyBoolean(), any()))
+                .willReturn(verificationStrategy);
+        attempt = prepareHtsAttemptWithSelectorAndCustomConfig(
+                RejectTokensTranslator.HRC_TOKEN_REJECT_FT,
+                subject,
+                enhancement,
+                addressIdConverter,
+                verificationStrategies,
+                gasCalculator,
+                configuration);
+
+        // when:
+        var call = subject.callFrom(attempt);
+
+        // then:
+        assertEquals(DispatchForResponseCodeHtsCall.class, call.getClass());
```

Comment:
Consider the `verify` condition as in the token cancel PR

Reply:
Added